src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -331,17 +331,16 @@ private static bool IsOSVersionAtLeast(int major, int minor, int build, int revi
             {
                 return current.Major > major;
             }
-            if (current.Minor != minor)
+            if (current.Minor >= 0 && current.Minor != minor)
             {
                 return current.Minor > minor;
             }
-            if (current.Build != build)
+            if (current.Build >= 0 && current.Build != build)
```

Comment:
```suggestion
            if (current.Build < 0)
            {
                // Unspecified component satisfies any required version
                return true;
            }
            if (current.Build != build)
```
We can exit early. We do not need to continue comparing Revision when Build was unspecified.